hpe3parclient/client.py

```diff
@@ -4816,3 +4816,28 @@ def getScheduleStatus(self, schedule_name):
                 return 'active'
         msg = "Couldn't find the schedule '%s' status" % schedule_name
         raise exceptions.SSHException(reason=msg)
+
+    @staticmethod
+    def getPortNumber(ip, login, password, port=22,
+                      conn_timeout=None, privatekey=None,
+                      **kwargs):
+        """Get port number from showwsapi output
+
+        :param 3PAR credentials
+        :return: 443: If port 443 is in showwsapi output
+        :        8080: If port 8080 is in showwsapi output
+        """
+        ssh_client = ssh.HPE3PARSSHClient(ip, login, password, port,
+                                          conn_timeout, privatekey,
+                                          **kwargs)
+        if ssh_client is None:
+            raise exceptions.SSHException('SSH is not initialized. Initialize'
+                                          ' it by calling "setSSHOptions".')
+        ssh_client.open()
+        wsapi_output = ssh_client.run(['showwsapi'])
+        if ssh_client:
+            ssh_client.close()
+        if '443' in wsapi_output[1]:
```

Comment:
@ansarars This logic also doesnt hold good. We must pick up the value from column "HTTPS_Port". Please try this.

Reply:
We get run([showwsapi]) output as a list of two members as given below, first element of the list is header and second element is values. We get 443 string in second  element of list hence I wrote "if '443' in list[1]:" logic:-
['-Service-,-State-,HTTPS_Port,-Version-,------------------API_URL-------------------', 'Enabled,Active,443,1.7.0,https://vp2-157.in.rdlabs.hpecorp.net/api/v1']